library/src/main/java/com/novoda/viewpageradapter/ViewPagerAdapterState.java

```diff
@@ -40,8 +40,11 @@ private static ViewPagerAdapterState from(Parcel in) {
     }
 
     private static SparseIntArray extractIdsFrom(Bundle bundle) {
-        SparseIntArray output = new SparseIntArray();
         int[] ids = bundle.getIntArray(KEY_VIEW_IDS);
+        if (ids == null) {
```

Comment:
It's a bundle, you never know what is in there. You cannot assume the one you get does contain the saved view IDs